hasura.planx.uk/migrations/1716902907392_alter_table_public_flows_add_column_name/up.sql

```diff
@@ -0,0 +1,9 @@
+alter table "public"."flows" add column "name" text
+ null;
+ comment on column "public"."flows"."name" is 'The name of the flow, entered by the user and used to generate the "slug"';
+
+UPDATE flows
+SET name = replace(initcap(replace(slug,'-','xax')),'xax',' ')
```

Comment:
Ah I figured `initcap()` would just handle the first letter, not the first letter of each word. This makes it trickier for sure!

Using `"xax"` is clever and certainly works as we don't have this substring in any flow slugs (tested with `select * from flows where slug ilike '%xax%';`

However it's a little bit unintuitive and took me a while to work out what was going on here 😅 It also relies on strings which users _could_ modify.

Is there another approach here that might be more robust? It should be possible to use substrings (e.g. upper case just first letter, concatenate with remaining lowercase string).

https://pgplayground.com/ can be handy as a quick playground for functions like this 👍

Reply:
Aha, so I should be explicitly altering the first letter as it'll always output the same result no matter what the user adds (unless they use dashes), and it's clearer to read?

Something like this:

```sql
SET name = replace(concat(upper(substring(slug,1,1)),right(slug,-1)),'-',' ')
```

And using the above ``UPPER`` method is easier to read than my previous method, so even changing out the ``xax`` string to use something like ``_`` wouldn't make it more robust, as shown below. 

```sql
SET name = replace(initcap(replace(slug,'-','_')),'_',' ')
```

I've been using the Hasura SQL console to try different methods since it required less setup... any concerns with that?

Reply:
No - totally legit and safe to use your local Hasura console for this also! 👌 

I'd favour the first option as despite being fairly complex syntax, the intention feels pretty clear.

However, the second approach would also work well and I have no objections! 👍 

As seen below, users cannot have special characters in the existing slug, so using the special character `_` as a replacement string is safe - 

![image](https://github.com/theopensystemslab/planx-new/assets/20502206/75d3271b-310a-4d49-9cdb-4026a66a8b2d)
![image](https://github.com/theopensystemslab/planx-new/assets/20502206/0eddff1b-9333-4737-b8a9-02606c93b7b5)

And however unlikely it is that they use the string `xax`, it's certainly possible, so it's best not to use this - 

![image](https://github.com/theopensystemslab/planx-new/assets/20502206/4795b244-ab9c-41dc-bfdd-e720adae95e7)
![image](https://github.com/theopensystemslab/planx-new/assets/20502206/fa9e3a8d-1d29-4ff0-9fe7-6db291aef961)